hibernate-core/src/main/java/org/hibernate/loader/ast/internal/AbstractMultiIdEntityLoader.java

```diff
@@ -39,14 +38,12 @@
 public abstract class AbstractMultiIdEntityLoader<T> implements MultiIdEntityLoader<T> {
 	private final EntityMappingType entityDescriptor;
 	private final SessionFactoryImplementor sessionFactory;
-	private final EntityIdentifierMapping identifierMapping;
-	protected final Object[] idArray;
+	protected final EntityIdentifierMapping identifierMapping;
 
 	public AbstractMultiIdEntityLoader(EntityMappingType entityDescriptor, SessionFactoryImplementor sessionFactory) {
 		this.entityDescriptor = entityDescriptor;
 		this.sessionFactory = sessionFactory;
 		identifierMapping = getLoadable().getIdentifierMapping();
-		idArray = (Object[]) Array.newInstance( identifierMapping.getJavaType().getJavaTypeClass(), 0 );
```

Comment:
This was the main problem affecting Quarkus, FWIW.
This being done in the abstract class meant it was done for `MultiIdEntityLoaderArrayParam` -- where it's needed, and generally innocuous since it handles things like `Integer[]` -- but also `MultiIdEntityLoaderStandardParam` -- where it's not needed, and possibly problematic as it could be handling things like `MyIdClass[]`, requiring `MyIdClass[]` (the array type) to be registered for reflection in native images.
It's not trivial to detect all ID types in Quarkus (think of `orm.xml` using property access, or IDs defined in mappedsuperclasses with generics), so this need for registering `MyIdClass[]` for reflection was a problem.